### PR TITLE
Bump PL to 0.46.0-dev94

### PR DESCRIPTION
**Pull request by maliasadi**
**Context:**
Bump PL version to `0.46.0-dev94` (latest) and fix the gradient tests with adjoint/ctrl rules

**Description of the Change:**

**Benefits:**
- Pass LLL :green_circle: 

**Possible Drawbacks:**

**Related GitHub Issues:**
[sc-130038]

### CODE REVIEW

**Review by mlxd — APPROVED** [review]

File: frontend/test/lit/test_gradient.py [hunk 1]
```diff
@@ -355,3 +355,87 @@ def cost(params: jax.core.ShapedArray([2], float)):
 
 
 print(test_non_diff_ops_in_cost_and_grad.mlir)
+
```
Comment (kipawaa) [inline frontend/test/lit/test_gradient.py:358]: We could probably remove the non-capture tests, since we only care about `capture=True` moving forwards

**Review by kipawaa — APPROVED** [review]
Wonderful, thanks @maliasadi

File: frontend/test/lit/test_gradient.py (hunk 1 above)
Comment (maliasadi) [reply frontend/test/lit/test_gradient.py:358]: I just added them for verbosity and consistency with the prev pytests.

**Review by mehrdad2m — APPROVED** [review]
Awesome! Thanks for the help @maliasadi @kipawaa
